lib/ex_doc/formatter/html.ex

```diff
@@ -602,7 +621,15 @@ defmodule ExDoc.Formatter.HTML do
 
       {path, opts} ->
         base = path |> to_string() |> Path.basename()
-        {base, opts[:filename] || Utils.text_to_id(Path.rootname(base))}
+
+        txid =
+          cond do
+            filename = opts[:filename] -> filename
+            url = opts[:url] -> url
```

Comment:
I am pretty sure you want to skip the `:url` in this case, because this is used to convert something like ``[foo](foo.md)`` into a link for `foo.html`, but there is nothing to link to in the URL case.